Library/Client.ts

```diff
@@ -131,17 +132,22 @@ class Client {
         this.track(data);
     }
 
-    public trackRequestSync(request: http.ServerRequest, response: http.ServerResponse, ellapsedMilliseconds?: number, properties?: { [key: string]: string; }, error?: any) {
-        RequestTracking.trackRequestSync(this, request, response, ellapsedMilliseconds, properties, error);
+    public trackServerRequestSync(request: http.ServerRequest, response: http.ServerResponse, ellapsedMilliseconds?: number, properties?: { [key: string]: string; }, error?: any) {
+        ServerRequestTracking.trackRequestSync(this, request, response, ellapsedMilliseconds, properties, error);
     }
 
-    public trackRequest(request: http.ServerRequest, response: http.ServerResponse, properties?: { [key: string]: string; }) {
-        RequestTracking.trackRequest(this, request, response, properties);
+    public trackServerRequest(request: http.ServerRequest, response: http.ServerResponse, properties?: { [key: string]: string; }) {
+        ServerRequestTracking.trackRequest(this, request, response, properties);
+    }
+
+    public trackClientRequest(requestOptions: any, request: http.ClientRequest, properties?: { [key: string]: string; }) {
+        ClientRequestTracking.trackRequest(this, requestOptions, request, properties);
     }
 
     public trackDependency(
         name: string,
         commandName: string,
+        target: string,
```

Comment:
The addition of the `target` parameter here is a breaking change. To avoid that it will need to be moved to an optional parameter at the end, although I think the intention of the new contract is actually that it should be required.